### ISSUE TITLE
authentication in LocalGemma2ForCausalLM with huggingface access token doesn't work

### ISSUE THREAD

**Issue by dsingal0** [huggingface/local-gemma#21]
Authentication in code with token=hf_token doesn't work unless you use subprocess.run("local-gemma", "--token", hf_token, "What is the capital of France")

`model = LocalGemma2ForCausalLM.from_pretrained("google/gemma-2-9b",  token=hf_token, preset="memory")
`

**Comment by sanchit-gandhi** [comment]
Hey @dsingal0! Unfortunately I'm not able to reproduce the error, e.g. using the following example bellow correctly uses the token I specify:

```sh
local-gemma --token "hf_..."         
```

The PR #22 should make logging in using your HF more straightforward!

**Comment by dsingal0** [comment]
I don't believe the cli has auth issues, but it didn't work in regular python using just LocalGemma2ForCausalLM. 

`model = LocalGemma2ForCausalLM.from_pretrained("google/gemma-2-9b",  token=hf_token, preset="memory")`

**Comment by dsingal0** [comment]
I can't do `huggingface-cli login` since I'm using baseten.co to deploy gemma2, and unless I store my HF API token as an envvar as opposed to a docker secret it isn't accessible before I'm in python code, so I'd have to do `subprocess.run`.

**Comment by ilarramendi** [comment]
`huggingface-cli login` worked for me, the token parameter seems to be not working

**Comment by sanchit-gandhi** [comment]
Hey @dsingal0 @ilarramendi - circling back here. Unfortunately, I'm not able to reproduce the error using the Python API. E.g. if I first verify that I'm not logged-in:
```sh
sanchit@vorace:~$ huggingface-cli whoami
Not logged in
```

And then running the following code-snippet, passing the token to `.from_pretrained`:
```python
from local_gemma import LocalGemma2ForCausalLM

model = LocalGemma2ForCausalLM.from_pretrained("google/gemma-2-9b", preset="memory", token="hf_")
```

I am able to download the model locally:
```
config.json: 100%|█████████████████████████████████████████████████████████████████████| 856/856 [00:00<00:00, 4.03MB/s]
model.safetensors.index.json: 100%|█████████████████████████████████████████████████| 39.1k/39.1k [00:00<00:00, 490kB/s]
...
```

Could you please:
1. Double check that you've created a token with the necessary permissions (read access of the Gemma-2 repos)
2. Past the full traceback of the error you're encountering (if 1 does not fix the issue)

Thanks!
